utils/Utils.java

```diff
@@ -66,11 +58,14 @@ public static double[] swerveModuleStatesToArray(SwerveModuleState[] states) {
         return array;
     }
 
-    /**
-     * Checks whether chassis speeds are epsilon equals to zero.
-     * @param speeds The chassis speeds to check.
-     * @return Whether the chassis speeds are epsilon equals to zero.
-     */
+    public static SwerveModuleState[] arrayToSwerveModuleStates(double[] states) { //TODO: check
+        SwerveModuleState[] swerveModuleStates = new SwerveModuleState[states.length / 2];
+        for (int i = 0; i < states.length; i += 2) {
+            swerveModuleStates[i / 2] = new SwerveModuleState(states[i + 1], new Rotation2d(states[i]));
+        }
+        return swerveModuleStates;
+    }
+
     public static boolean speedsEpsilonEquals(ChassisSpeeds speeds) {
```

Comment:
This function isn't used anywhere. Also, if it's needed, it's better to put it in the swerve package and not in this global location.